### DISCUSSION TITLE
NULL in void WbSolid::setSolidMerger()

### DISCUSSION THREAD

**Discussion by ShuffleWire** [cyberbotics/webots#4063] [Q&A]
as in https://github.com/cyberbotics/webots/pull/4055
is it expected that, in WbSolid.cpp, void WbSolid::setSolidMerger()
we do 
    mSolidMerger = NULL;
?
i think we should do 
   mSolidMerger.clear()

**Comment by omichel** [comment]
Both are exactly equivalent from a binary point of view, however `mSolidMerger.clear()` is better because it shows this is not a standard C++ pointer, but a `QPointer`, so this is more readable.

**Reply by ShuffleWire** [reply]
that's true, the doc state that, sorry...
https://doc.qt.io/qt-5/qpointer.html#operator-eq
I believe leave it like that will cause no harm, the QPointer is made for this kind of usage !
